DataFormats/PatCandidates/src/PackedCandidate.cc

```diff
@@ -105,6 +108,43 @@ void pat::PackedCandidate::unpackCovariance() const {
     unpackCovarianceElement(*m, packedCovariance_.dxydz, 3, 4);
     unpackCovarianceElement(*m, packedCovariance_.dlambdadz, 1, 4);
     unpackCovarianceElement(*m, packedCovariance_.dphidxy, 2, 3);
+
+    if (forcePosDef) {
+      //      std::cout<<"unpacking enforcing positive-definite cov matrix"<<std::endl;
+      //calculate the determinant and verify positivity
+      double det_(0);
+      bool computed_ = m->Det(det_);
+      //      std::cout<<"during unpacking, the determinant is: "<<det_<<std::endl;
+      if (computed_ && det_ < 0) {
```

Comment:
in mkFit context we apply a full Sylvester's criterion; computing determinants for sub-matrices. The extra cost is not that much compared to computing the det for the full matrix
https://github.com/cms-sw/cmssw/blob/17333a928fd7b69adc42695086b04fd8a55564ce/RecoTracker/MkFit/plugins/MkFitOutputConverter.cc#L321-L338

Reply:
thanks for the pointer indeed.